Description: Add failsafe options for acpi and nouveau on boot screen
Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Created: 2019-03-20
Last-Update: 2022-08-07

--- grub2-2.06.orig/util/grub.d/10_linux.in
+++ grub2-2.06/util/grub.d/10_linux.in
@@ -130,6 +130,10 @@ linux_entry ()
 	      title="$(gettext_printf "%s, with Linux %s (%s)" "${os}" "${version}" "$(gettext "${GRUB_RECOVERY_TITLE}")")" ;;
 	  init-*)
 	      title="$(gettext_printf "%s, with Linux %s (%s)" "${os}" "${version}" "${type#init-}")" ;;
+	  nouveau-off)
+	      title="$(gettext_printf "%s, with Linux %s (%s)" "${os}" "${version}" "$(gettext "Nouveau/Nvidia Off")")" ;;
+      acpi-off)
+          title="$(gettext_printf "%s, with Linux %s (%s)" "${os}" "${version}" "$(gettext "ACPI Off")")" ;;
 	  *)
 	      title="$(gettext_printf "%s, with Linux %s" "${os}" "${version}")" ;;
       esac
@@ -380,6 +384,11 @@ while [ "x$list" != "x" ] ; do
     linux_entry "${OS}" "${version}" simple \
     "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
 
+    linux_entry "${OS}" "${version}" nouveau-off \
+              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} nouveau.modeset=0"
+    linux_entry "${OS}" "${version}" acpi-off \
+              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} acpi=off"
+
     submenu_indentation="$grub_tab"
     
     if [ -z "$boot_device_id" ]; then

